app/routes/sites/network-sites/network-site/buildings/new.js

```diff
@@ -0,0 +1,52 @@
+import Ember from 'ember';
+
+const {
+  RSVP
+} = Ember;
+
+export default Ember.Route.extend({
+  model() {
+    return RSVP.hash({
+      site: this.modelFor('sites.network-sites.network-site'),
+      buildings: this.modelFor('sites.network-sites.network-site').get('buildings'),
+      building: this.store.createRecord('building')
+    });
+  },
+
+  actions: {
+    addBuildingMarker(event) {
+      let latLng = event.latLng;
```

Comment:
You should receive just the value here. You can use `{{action 'actionName' value="target.value"}}` to make sure that you get the value and not the event.